### PR TITLE
P20-903: Display U13 Warning on Section Creation to USA teachers

### PR DESCRIPTION
**Pull request by artem-vavilov**
## Error
```
==============[0m[1000D[KERROR["test_returns_nil_if_student_us_state_is_\"??\"", "UserTest::UsStateCodeTest", 56.4121803099988]
 test_returns_nil_if_student_us_state_is_"??"#UserTest::UsStateCodeTest (56.41s)
Minitest::UnexpectedError:         Errno::EACCES: Permission denied @ rb_sysopen - /usr/local/lib/ruby/gems/3.0.0/gems/city-state-1.1.0/lib/db/states.us
            app/models/user.rb:2746:in `us_state_code'
            test/models/user_test.rb:18:in `block in <class:UsStateCodeTest>'
            test/testing/setup_all_and_teardown_all.rb:36:in `run'
```
## Links
- JIRA ticket: [P20-903](https://codedotorg.atlassian.net/browse/P20-903)
- Related PR: https://github.com/code-dot-org/code-dot-org/pull/58497
- Revert PR: https://github.com/code-dot-org/code-dot-org/pull/58670

### CODE REVIEW

File: dashboard/test/models/user_test.rb [hunk 1]
```diff
@@ -7,6 +7,28 @@ class UserTest < ActiveSupport::TestCase
...
+
+    test 'returns teacher school US state code' do
+      teacher = create(:teacher, school_info: create(:school_info, state: 'ny'))
+      assert_equal 'NY', teacher.us_state_code
+    end
+
+    test 'returns teacher school US state code when state is name' do
+      teacher = create(:teacher, school_info: create(:school_info, state: 'washington dc'))
+      assert_equal 'DC', teacher.us_state_code
+    end
```
Comment (wilkie) [inline dashboard/test/models/user_test.rb:29]: Maybe we should cover the case where the `school_info` does not have a state recorded or has an unknown state ... maybe it is a school in Canada, Europe, etc. I think you handle that case in your implementation.

Comment (wilkie) [reply dashboard/test/models/user_test.rb:29]: Basically, cover the case where the teacher's state would be `nil` I think.

Comment (artem-vavilov) [reply dashboard/test/models/user_test.rb:29]: It stores only U.S. states

Comment (artem-vavilov) [reply dashboard/test/models/user_test.rb:29]: https://github.com/code-dot-org/code-dot-org/pull/58688/commits/e187410d8cb2af0c28f764763d420bca925c9955

**Review by wilkie — APPROVED** [review]
Great work! 👍

**Pull request merged**